build_danm.sh

```diff
@@ -30,12 +53,78 @@ error_handler()
   exit $1
 }
 
-if [[ ( "$TRAVIS_PIPELINE" = "buildah" ) || ( "$TRAVIS_PIPELINE" = ""  && -x "$(command -v buildah)" ) ]]; then
-  source ./build_buildah.sh
-elif [[ ( "$TRAVIS_PIPELINE" = "docker" ) || ( "$TRAVIS_PIPELINE" = ""  && -x "$(command -v docker)" ) ]]; then
-  source ./build_docker.sh
+
+#
+# Don't use cache unless we're told otherwise.
+#
+if [ -z "${USE_CACHE}" ]
+then
+  FIRST_BUILD_EXTRA_BUILD_ARGS="--no-cache"
+fi
+
+
+#
+# Identify if we need to run docker or buildah
+#
+if [[ ( "$TRAVIS_PIPELINE" = "buildah" ) || ( "$TRAVIS_PIPELINE" = ""  && -x "$(command -v buildah)" ) ]]
+then
+  BUILD_COMMAND="buildah bud"
+  TAG_COMMAND="buildah tag"
+elif [[ ( "$TRAVIS_PIPELINE" = "docker" ) || ( "$TRAVIS_PIPELINE" = ""  && -x "$(command -v docker)" ) ]]
+then
+  BUILD_COMMAND="docker image build"
+  TAG_COMMAND="docker image tag"
 else
  echo 'The build process requires docker or buildah/podman installed. Please install any of these and make sure these are executable'
  exit 1
 fi
 
+
+#
+# Construct a unique version number from the git commit
+# hash that is being build. If the workspace isn't
+# clean (ie. if "git status" would say anything other than
+# "working tree clean"), add a _dirty suffix to that
+# version number.
+#
+LATEST_TAG=$(git describe --tags)
+COMMIT_HASH=$(git rev-parse --short=8 HEAD)
+if [ -n "$(git status --porcelain)" ]
+then
+  COMMIT_HASH="${COMMIT_HASH}_dirty"
+fi
+
+
+#
+# Determine which build stages we want to tag as an image.
+#
+build_targets=(netwatcher svcwatcher webhook danm-cni-plugins)
+
+if [ -n "${KEEP_BUILDER}" ]
+then
+    build_targets+=(builder)
+fi
+
+#
+# Build the various images. Each image is represented
+# by one target in the multi-stage Dockerfile.
+#
+for plugin in ${build_targets[@]}
+do
+  echo Building: ${plugin}, version ${COMMIT_HASH}
+  ${BUILD_COMMAND} \
+    ${EXTRA_BUILD_ARGS} \
+    ${FIRST_BUILD_EXTRA_BUILD_ARGS} \
+    --build-arg LATEST_TAG=${LATEST_TAG} \
+    --build-arg COMMIT_HASH=${COMMIT_HASH} \
```

Comment:
`--build-arg` can be specified with variable name only, like `--build-arg LATEST_TAG` and its value will be loaded from the environment. I think it would be cleaner this way

Reply:
Could do, but keep in mind that that would require `export`ing these variables from the build script. It certainly would be another option -- not sure if it'd be all that much better though.

Reply:
Okay, then leave it as is.